src/components/SUE/report/SueReportLocation.tsx

```diff
@@ -177,6 +177,71 @@ export const SueReportLocation = ({
     return <LoadingSpinner loading />;
   }
 
+  const onMapPress = async ({
+    nativeEvent
+  }: {
+    nativeEvent: { action: string; coordinate: Location.LocationObjectCoords };
+  }) => {
+    if (nativeEvent.action !== 'marker-press' && nativeEvent.action !== 'callout-inside-press') {
+      setUpdatedRegion(false);
+      setUpdateRegionFromImage(false);
+      setSelectedPosition(nativeEvent.coordinate);
+
+      try {
+        await handleGeocode(nativeEvent.coordinate);
+      } catch (error) {
+        setSelectedPosition(undefined);
+        Alert.alert(texts.sue.report.alerts.hint, error.message);
+        throw new Error(error.message);
+      }
+    }
+  };
+
+  const onMyLocationButtonPress = async (isFullScreenMap = false) => {
+    if (!isFullScreenMap) {
+      Alert.alert(texts.sue.report.alerts.hint, texts.sue.report.alerts.myLocation, [
+        {
+          text: texts.sue.report.alerts.no
+        },
+        {
+          text: texts.sue.report.alerts.yes,
+          onPress: async () => {
+            const location = position || lastKnownPosition;
+
+            if (location) {
+              setUpdatedRegion(true);
+              setUpdateRegionFromImage(false);
+              setSelectedPosition(location.coords);
+
+              try {
+                await handleGeocode(location.coords);
+              } catch (error) {
+                setSelectedPosition(undefined);
+                Alert.alert(texts.sue.report.alerts.hint, error.message);
+              }
+            }
+          }
+        }
+      ]);
+    } else {
+      const location = position || lastKnownPosition;
+
+      if (location) {
+        setUpdatedRegion(true);
+        setUpdateRegionFromImage(false);
+        setSelectedPosition(location.coords);
+
+        try {
+          await handleGeocode(location.coords);
+        } catch (error) {
+          setSelectedPosition(undefined);
+          Alert.alert(texts.sue.report.alerts.hint, error.message);
+          throw new Error(error.message);
```

Comment:
why is this throw here but not in the part for the small map?

Reply:
Because I want to be able to recognise an error in `SueMapViewScreen` and make `setPosition` `undefined`. No need to create errors in the small map view. Because the `setSelectedPosition` value is updated as `undefined` in the catch section.